### ISSUE TITLE
Data loading issue on fresh node for larger datasets

### ISSUE THREAD

**Issue by FiV0** [xtdb/crux#2769]
The following test shows that there seems to be some data loading issue with the liveTrie when loading a fresh node. 
```clj
(deftest loading-issue-on-live-trie
  
  (let [node-dir (io/file "dev/loading-issue")]
    ;; can be commented out after first run 
    (with-open [node (tu/->local-node {:node-dir (.toPath node-dir)})]
      ;; you might need to push this number up if the issue does not show up on your machine
      (doseq [tx (->> (range 10000)
                      shuffle
                      (map (comp #(vector :put :xt-docs %) #(hash-map :xt/id % :s (str %))))
                      (partition-all 1024))]
        (xt/submit-tx node tx))
      ;; assure that largest id is in the live trie
      (xt/submit-tx node [[:put :xt-docs {:xt/id 100000000 :s "100000000"}]])

      ;; this makes things pass 
      #_(tu/finish-chunk! node)
      )
    
    (with-open [node (tu/->local-node {:node-dir (.toPath node-dir)})]
      ;; this makes things pass
      #_(Thread/sleep 100)
      (t/is (= [{:i 100000000}] 
               (xt/q node '{:find [i]
                            :where [(match :xt-docs {:xt/id i})]
                            :order-by [[i :desc]]
                            :limit 1}))))))
```

**Comment by FiV0** [comment]
The issue that is happening here is that the log contains a bunch a of data which doesn't yet make up a whole chunk, hence no metadata about chunks has been written. When the `indexer` gets initialized it is done so with a `latest-completed-tx` of `nil` (the indexer doesn't know yet anything about the new stuff coming through the log). The query then runs on empty data or only partial data and only after a short initialization phase the indexer is up to date. To make this consistent we need to add something like a `latest-seen-tx` or similar to the indexer that is independent of `latest-completed-tx`.

**Comment by FiV0** [comment]
Closing in favor of #2789.
